src/main/java/org/opentripplanner/routing/algorithm/postprocessing/elements/ItineraryPostProcessingElement.java

```diff
@@ -0,0 +1,35 @@
+package org.opentripplanner.routing.algorithm.postprocessing.elements;
+
+import org.opentripplanner.model.plan.Itinerary;
+
+import java.util.List;
+
+/**
+ * Filter, modify or sort itineraries. A post-processing element can modify the elements in the
+ * list, but not the List passed in. It should treat the list as immutable. Do not change the list
+ * passed into the element, instead make a copy, change it and return the copy. It is allowed to
+ * return the list unchanged.
+ * <p>
+ * A post-processing element should do only one thing! For example do not change the itineraries
+ * and delete itineraries in the same post-processing element. Instead, create an enhancer and a
+ * filter and insert them after each other in the post-processor.
+ * <p>
+ * This allows decoration of each post-processing element and makes it easier to reuse logic. Like
+ * the {@link org.opentripplanner.routing.algorithm.postprocessing.filters.MaxLimitFilter} is
+ * reused in several places.
+ */
+public interface ItineraryPostProcessingElement {
```

Comment:
I think `ItineraryPostProcessessor` is a better name, indicating that this is doing something - rater than holding information.